bugwarrior/services/github.py

```diff
@@ -278,29 +293,34 @@ def get_involved_issues(self, user):
         """ Grab all 'interesting' issues """
         issues = {}
         for issue in self.client.get_involved_issues(user):
-            url = issue['html_url']
-            tag = re.match('.*github\\.com/(.*)/(issues|pull)/[^/]*$', url)
-            if tag is None:
-                log.critical(" Unrecognized issue URL: %s.", url)
-                continue
-            issues[url] = (tag.group(1), issue)
+            try:
+                url = issue['html_url']
+                repo = self.get_repository_from_issue(issue)
+                issues[url] = (repo, issue)
+            except ValueError as e:
+                log.critical(e)
         return issues
 
     def get_directly_assigned_issues(self):
-        project_matcher = re.compile(
-            r'.*/repos/(?P<owner>[^/]+)/(?P<project>[^/]+)/.*'
-        )
         issues = {}
         for issue in self.client.get_directly_assigned_issues():
-            match_dict = project_matcher.match(issue['url']).groupdict()
-            issues[issue['url']] = (
-                '{owner}/{project}'.format(
-                    **match_dict
-                ),
-                issue
-            )
+            repos = self.get_repository_from_issue(issue)
+            issues[issue['url']] = (repos, issue)
         return issues
 
+    @staticmethod
```

Comment:
I see what you're saying now. I agree that calling the class by its explicit name in a static method would almost always be bad, but I disagree that there's no good reason to use them.

A `staticmethod` indicates to developers that the method probably does not access other class methods, class state, or instance state. I find this a helpful part of the method signature in indicating the function's purpose and role in the class.

Generally, `staticmethod` indicates that the method could just as well be a free function, but it is part of the class for organizational purposes. When a class has a lot of `staticmethod`'s, it's usually a sign that a lot of the methods would be better moved into a different module as free functions. With `classmethod`'s, this anti-pattern is much less obvious.